src/realm/query.hpp

```diff
@@ -171,6 +171,15 @@ class Query final {
     Query& less_equal(size_t column_ndx, Timestamp value);
     Query& less(size_t column_ndx, Timestamp value);
 
+    // Conditions: size
+    Query& size_equal(size_t column_ndx, int64_t value);
+    Query& size_not_equal(size_t column_ndx, int64_t value);
+    Query& size_greater(size_t column_ndx, int64_t value);
+    Query& size_greater_equal(size_t column_ndx, int64_t value);
+    Query& size_less_equal(size_t column_ndx, int64_t value);
+    Query& size_less(size_t column_ndx, int64_t value);
+    Query& size_between(size_t column_ndx, int64_t from, int64_t to);
```

Comment:
I'm not sure it's worth adding these to the older `Query` interface. Most new query functionality is only added at the query expression level, since that's the only place that deals with all of the complexities that bindings need (traversing links, for example).